### ISSUE TITLE
PKGBUILD for us Arch users

### ISSUE THREAD

**Issue by ErikReider** [WillPower3309/swayfx#12]
- **Description:**
A regular PKGBUILD would be cool for us who want to easily test this cool fork :)

Editing the regular sway-git PKGBUILD would also work.

Btw, kinda unrelated but what's the difference between this and sway-borders? Both of these projects want to achieve the same goal

**Comment by WillPower3309** [comment]
Thanks for raising the issue! A PKGBUILD will be supplied once rounded corners is merged and blur is implemented (which should be a half week or so), I'll add this to the initial release milestone. The majority of the work was in making the renderer itself which is complete, however functionally near identical to the original wlr_renderer used in sway.

sway-borders is a fantastic repo, which takes a different type approach to rounded corners and borders. I wanted this repo to focus on rendering the rounded corners within the window itself, instead of appling them around the window. The sway-borders implementation is likely faster and allows for more customization, but rounding windows "internally" is much easier to make look good on a number of windows regardless of their color (as in sway-borders, you must match the corner / border colors), as well as being simpler to configure. I had this repo handy for the minimize scratchpad patch, and hadn't wanted to work around the existing rounded corner implementation in sway-borders.

The sway-borders maintainer and I are actually in talks now of collaboration, as they were also working on "internal" rounded corners. Stay tuned!

**Comment by ErikReider** [comment]
These PKGBUILD changes worked for me:

```diff
diff --git a/PKGBUILD b/PKGBUILD
index ccba678..0123b22 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -19,9 +19,9 @@ depends=(
 	"pango"
 	"polkit"
 	"pcre2"
-	"swaybg-git"
+	"swaybg"
 	"ttf-font"
-	"wlroots-git"
+	"wlroots"
 	"xorg-server-xwayland"
 )
 optdepends=(
@@ -39,7 +39,7 @@ optdepends=(
 backup=(etc/sway/config)
 arch=("i686" "x86_64")
 url="https://swaywm.org"
-source=("${pkgname%-*}::git+https://github.com/swaywm/sway.git"
+source=("${pkgname%-*}::git+https://github.com/WillPower3309/swayfx.git"
 	50-systemd-user.conf)
 sha512sums=('SKIP'
             '57590bc0d14c87289a4a9cd67991c6a841e54244d2a6186b5da5a08e633de2e8631959fa8c77ede211b0a5f315d920f2c1350951a53d6f2e9e81859056cb3c9e')

```

**Comment by WillPower3309** [comment]
> These PKGBUILD changes worked for me:
> 
> ```diff
> diff --git a/PKGBUILD b/PKGBUILD
> index ccba678..0123b22 100644
> --- a/PKGBUILD
> +++ b/PKGBUILD
> @@ -19,9 +19,9 @@ depends=(
>  	"pango"
>  	"polkit"
>  	"pcre2"
> -	"swaybg-git"
> +	"swaybg"
>  	"ttf-font"
> -	"wlroots-git"
> +	"wlroots"
>  	"xorg-server-xwayland"
>  )
>  optdepends=(
> @@ -39,7 +39,7 @@ optdepends=(
>  backup=(etc/sway/config)
>  arch=("i686" "x86_64")
>  url="https://swaywm.org"
> -source=("${pkgname%-*}::git+https://github.com/swaywm/sway.git"
> +source=("${pkgname%-*}::git+https://github.com/WillPower3309/swayfx.git"
>  	50-systemd-user.conf)
>  sha512sums=('SKIP'
>              '57590bc0d14c87289a4a9cd67991c6a841e54244d2a6186b5da5a08e633de2e8631959fa8c77ede211b0a5f315d920f2c1350951a53d6f2e9e81859056cb3c9e')
> ```
> 
> _Edited the regular sway-git PKGBUILD before installing_

Great work! Rounded corners are now merged, once blur is done I'll use this PKGBUILD!

**Comment by ErikReider** [comment]
The rounded corners seem to work just fine! :)

The only issues that I'm experiencing is the lack of AA (mentioned before by sway-borders maintainer) and the border radius which is mentioned in another issue

**Comment by WillPower3309** [comment]
> The rounded corners seem to work just fine! :)
> 
> The only issues that I'm experiencing is the lack of AA (mentioned before by sway-borders maintainer) and the border radius which is mentioned in another issue

AA will be added, I'll make an issue for it now! I'm going to implement blur first (just because I'm too excited about it to wait haha), but will add AA prior to release

**Comment by WillPower3309** [comment]
I lied, blur is frustrating. AA implementation in #18 

The corners seem a little off so I'll give it another look over prior to merging

**Comment by WillPower3309** [comment]
@ErikReider once 0.1 is released, would you be comfortable maintaining this PKGBUILD on the AUR?

**Comment by ErikReider** [comment]
I'd be happy to :)

**Comment by WillPower3309** [comment]
That's huge! Thanks a ton for the work there. I'm using nixOS, and already have a build working there! Looks like we'll have good coverage upon release
